Classes/Domain/Model/Changes/Property.php

```diff
@@ -144,163 +148,195 @@ public function canApply(): bool
-                ->getWorkspaceFinder()->findOneByCurrentContentStreamId($subject->subgraphIdentity->contentStreamId);
-            if (!$workspace) {
-                throw new \Exception(
-                    'Could not find workspace for content stream "' . $subject->subgraphIdentity->contentStreamId->value . '"',
-                    1699008140
-                );
-            }
+        if (is_null($subject) || is_null($propertyName) || $this->canApply() === false) {
+            return;
+        }
+
+        $workspace = $this->contentRepositoryRegistry->get($this->subject->subgraphIdentity->contentRepositoryId)
+            ->getWorkspaceFinder()->findOneByCurrentContentStreamId($subject->subgraphIdentity->contentStreamId);
+        if (!$workspace) {
+            throw new \Exception(
+                'Could not find workspace for content stream "' . $subject->subgraphIdentity->contentStreamId->value . '"',
+                1699008140
+            );
+        }
+
+        $commandResult = match (true) {
+            $this->getNodeType($subject)->hasReference($propertyName) => $this->handleNodeReferenceChange($workspace, $subject, $propertyName),
+            $propertyName === '_nodeType' => $this->handleNodeTypeChange($workspace, $subject, $propertyName),
+            $propertyName === '_hidden' => $this->handleHiddenPropertyChange($workspace, $subject, $propertyName),
+            default => $this->handlePropertyChange($workspace, $subject, $propertyName)
+        };
+
+        $commandResult->block();
+        $this->createFeedback($subject);
+    }
+
+    private function createFeedback(Node $subject): void
+    {
+        $propertyName = $this->getPropertyName();
 
-            if ($this->getNodeType($subject)->hasReference($propertyName)) {
-                // Use extra commands for reference handling
-                $value = $this->getValue();
-
-                $destinationNodeAggregateIds = [];
-
-                if (is_string($value) && !empty($value)) {
-                    $destinationNodeAggregateIds = [$value];
-                } elseif (is_array($value)) {
-                    $destinationNodeAggregateIds = $value;
-                }
-
-                $commandResult = $contentRepository->handle(
-                    SetNodeReferences::create(
-                        $workspace->workspaceName,
-                        $subject->nodeAggregateId,
-                        $subject->originDimensionSpacePoint,
-                        ReferenceName::fromString($propertyName),
-                        NodeReferencesToWrite::fromNodeAggregateIds(NodeAggregateIds::fromArray($destinationNodeAggregateIds))
-                    )
-                );
+        // !!! REMEMBER: we are not allowed to use $node anymore,
+        // because it may have been modified by the commands above.
+        // Thus, we need to re-fetch it (as a workaround; until we do not need this anymore)
```

Comment:
```suggestion
        // We have to refetch the Node after modifications because its a read-only model
        // These 'Change' classes have been designed with mutable Neos < 9 Nodes and thus this might seem hacky 
        // When fully redesigning the Neos Ui php integration this will fixed
```